core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3220,86 +3169,136 @@ private boolean syncMetadata(JournalContext journalContext, LockedInodePath inod
     return true;
   }
 
-  private boolean syncChildrenMetadata(JournalContext journalContext, LockedInodePath inodePath,
-      DescendantType syncDescendantType, BlockDeletionContext blockDeletionContext)
-      throws FileDoesNotExistException, InvalidPathException, IOException,
-      DirectoryNotEmptyException {
-    if (syncDescendantType == DescendantType.NONE) {
-      return false;
+  /**
+   * This class represents the result for a sync.
+   */
+  private static class SyncResult {
+    private boolean mLoadMetadata;
+    private boolean mDeletedInode;
+
+    SyncResult() {
```

Comment:
we can use a static factory method to give this constructor a name

Reply:
Done.